### PR TITLE
Add a tracker for the head position to update the XRCamera3D Node

### PR DESCRIPTION
**Pull request by patrickdown**
Add a tracker for the head position to update the XRCamera3D Node.  The tracker name is set to "_{glasses_id}_/head"

This can't be used as of yet because the XRCamera3D tracker name is hard coded to be "head".

### CODE REVIEW

**Comment by BastiaanOlij** [comment]
Ah, that looks good!

So the idea I had here is that we create a new class, `TiltFiveCamera3D` which inherits from `Camera3D`. This can then do an implementation similar to `XRCamera3D`:
https://github.com/godotengine/godot/blob/master/scene/3d/xr_nodes.cpp#L39

In Godot 3 `ARVRCamera` did a bunch more but nowadays all it does different from `Camera3D` is adding the tracking logic. We can make a slightly modified version that allows us to select for which glasses we're tracking (or build in the logic it gets it automatically based on the viewport it's apart of).

**Comment by patrickdown** [comment]
Okay this now implements a whole new set of T5 nodes that replace the XR ones.

File: extension/src/GodotT5Glasses.cpp [hunk 1]
```diff
@@ -84,7 +84,47 @@ PackedFloat64Array GodotT5Glasses::get_projection_for_eye(Glasses::Eye view, dou
     return arr;    
 }
 
+void GodotT5Glasses::on_glasses_reserved() {
+	XRServer *xr_server = XRServer::get_singleton();
+	ERR_FAIL_NULL(xr_server);
+
+    auto tracker_name = std::format("{}/head", get_id());
```
Comment (BastiaanOlij) [inline extension/src/GodotT5Glasses.cpp:91]: nitpicking but to be in line with OpenXR, which I think is a good long term strategy, I'd go for `/user/{}/head`

**Review by BastiaanOlij — APPROVED** [review]
I wonder if adding `T5Node3D` and `T5Controller3D` isn't overkill, we're adding Origin and Camera because we need logic that is distinctly different from their XR counterparts, but for node and controller we can use the build in ones and piggy back on any future fixes in core, it seems wasteful to duplicate them?

**Pull request merged**